File: day 79/homework/task1.py
import math

def count_area(shape, *dimensions):
    if shape == "rectangle":
        if len(dimensions) == 2:
            width, height = dimensions
            return width * height
        else:
            return "invalid input for rectangle"
    
    elif shape == "circle":
        if len(dimensions) == 1:
            radius = dimensions[0]
            return math.pi * radius ** 2
        else:
            return "invalid input for circle"
    
    elif shape == "triangle":
        if len(dimensions) == 3:
            base, height, _ = dimensions
            return 0.5 * base * height
        else:
            return "invalid input for triangle"
    
    else:
        return "unrecongizable shape"
    
    
# ----------------------------------------------------

print(count_area("rectangle", 5, 10))

print(count_area("circle", 5)) 

print(count_area("triangle", 10, 5, 6))

File: day 079/homework/task1.js
function countArea(shape, ...dimensions) {
    if (shape === "rectangle") {
        if (dimensions.length === 2) {
            let [width, height] = dimensions;
            return width * height;
        } else {
            return "invalid input (rectangle)";
        }
    } 
    else if (shape === "circle") {
        if (dimensions.length === 1) {
            let radius = dimensions[0];
            return Math.PI * Math.pow(radius, 2);
        } else {
            return "invalid input (circle)";
        }
    } 
    else if (shape === "triangle") {
        if (dimensions.length === 3) {
            let [base, height, _] = dimensions;
            return 0.5 * base * height;
        } else {
            return "invalid input (triangle)";
        }
    } 
    else {
        return "shape not recognized";
    }
}

// ---------------------------------------------------------

console.log(countArea("rectangle", 5, 10));

console.log(countArea("circle", 7));

console.log(countArea("triangle", 10, 5, 6));
